tests/test_directives.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) 2017 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+import sys
+
+from flow import Directives, _DirectivesItem, OnlyType
+
+
+def raise_if_too_small(value):
```

Comment:
I believe you asked @bdice a question regarding this, but why are `raise_if_too_small`, `_evaluate`,  `finalize_np`,  `is_fraction`, and `no_aggregation` copied ?

Reply:
This was done in order to prevent a lot of importing in `flow/__init__.py`. This was a temporary solution as `from . import directives as directives_module` was referring to the `directives` class in `operations.py` module

Reply:
We need to import them, otherwise we won't know if they get broken in the flow/directives.py file. I much prefer the from module import name as new name approach.